src/System.Drawing.Common/src/System/Drawing/Internal/SystemColorTracker.cs

```diff
@@ -0,0 +1,163 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#if FEATURE_SYSTEM_EVENTS
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.InteropServices;
+using Microsoft.Win32;
+
+namespace System.Drawing.Internal
+{
+    // Keeps track of objects that need to be notified of system color change events.
+    // Mostly this means maintaining a list of weak references.
+    internal class SystemColorTracker
+    {
+        // when I tried the self host, it went over 500 but never over 1000.
+        private static int INITIAL_SIZE = 200;
+        // If it gets this big, I seriously miscalculated the performance of this object.
+        private static int WARNING_SIZE = 100000;
+        private static float EXPAND_THRESHOLD = 0.75f;
+        private static int EXPAND_FACTOR = 2;
+
+        private static WeakReference[] list = new WeakReference[INITIAL_SIZE];
+        private static int count = 0;
+        private static bool addedTracker;
+        private static object lockObject = new object();
+
+        // There's no such thing as a delegate to a static method,
```

Comment:
You can create delegates over static methods...

Reply:
That's not my comment.  I suppose I can just delete it, I'll try to figure out what they were talking about and make it more correct.